packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmInMemoryTests.kt

```diff
@@ -0,0 +1,243 @@
+
+        val conf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(fileName)
+            .build()
+        val encConf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(encFileName)
+            .encryptionKey(key)
+            .build()
+        Realm.deleteRealm(conf)
+        Realm.deleteRealm(encConf)
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { stringField = "foo" })
+        }
+
+        // Tests a normal Realm file.
+        realm.writeCopyTo(conf)
+        val onDiskRealm: Realm = Realm.open(conf)
+        assertEquals(1, onDiskRealm.query<Sample>().count().find())
+        onDiskRealm.close()
+
+        // Tests a encrypted Realm file.
+        realm.writeCopyTo(encConf)
+        val onDiskEncryptedRealm: Realm = Realm.open(encConf)
+        assertEquals(1, onDiskEncryptedRealm.query<Sample>().count().find())
+        onDiskEncryptedRealm.close()
+
+        // Tests with a wrong key to see if it fails as expected.
+        val randomKey = Random.nextBytes(64)
+        RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(encFileName)
+            .encryptionKey(randomKey)
+            .build()
+            .let { conf ->
+                assertFailsWith(IllegalArgumentException::class, "Encrypted Realm should not be openable with a wrong encryption key") {
+                    Realm.open(conf)
+                }
+            }
+    }
+
+    // Tests writeCopyTo result when called in a transaction.
+    @Test
+    fun writeCopyToInTransaction() {
+        val fileName: String = tmpDir + ".realm"
+        val conf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(fileName)
+            .build()
+        Realm.deleteRealm(conf)
+        lateinit var onDiskRealm: Realm
+        // The value insertion is not completed until after the scope of writeBlocking, meaning that both
+        // querys result will be 0
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { stringField = "foo" })
+            realm.writeCopyTo(conf)
+            onDiskRealm = Realm.open(conf)
+            assertEquals(0, onDiskRealm.query<Sample>().count().find())
+        }
+        assertEquals(0, onDiskRealm.query<Sample>().count().find())
+        onDiskRealm.close()
+    }
+
+    // Test below scenario:
+    // 1. Creates a in-memory Realm instance in the main thread.
+    // 2. Creates a in-memory Realm with same name in another thread.
+    // 3. Closes the in-memory Realm instance in the main thread and the Realm data should not be released since
+    //    another instance is still held by the other thread.
+    // 4. Closes the in-memory Realm instance and the Realm data should be released since no more instance with the
+    //    specific name exists.
+    @Test
+    @Throws(InterruptedException::class, ExecutionException::class)
+    fun multiThread() {
```

Comment:
This test looks very Java-like, and we could probably rewrite it so it uses coroutines instead. That said, if it runs fine on both JVM and Native keeping as is, is probably fine for now.